### PR TITLE
DAOS-8530 doc: add documentation to explain auto class selection

### PR DESCRIPTION
**Pull request by mchaarawi**
Skip-build: true
Skip-test: true

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/object/README.md [hunk 1]
```diff
@@ -34,11 +34,66 @@ In addition user can register customized object class by
...
+Users can select the object class manually when generating the oid from the list of all object
+classes in [/src/include/daos_obj_class.h]. However manually selecting the object class is not
+encouraged for regular users and should be done by advanced users only who understand all the
+different object classes and the redundancy factor of the container. For most users, passing an
+OC_UNKNOWN (0) object class to `daos_obj_generate_oid()` would allow DAOS to automatically select an
+object class based on the container properties where that object is being accessed such as the
+redundancy factor (RF), the number of domain (server engines) of the pool, and on the type of object
+being accessed (determined by the feats flag).
+
+The following details how the object class is chosen when no default or hints are provided:
```
Comment (shimizukko) [inline src/object/README.md:51]:
```suggestion
The following details show how the object class is chosen when no default or hints are provided:
```
typo?

File: src/object/README.md [hunk 2]
```diff
@@ -34,11 +34,66 @@ In addition user can register customized object class by
...
+  - no feats type: OC_RP_3G1
+- RF:3
+  - Array, Byte Array, Flat KV object: OC_RP_4GX
+  - no feats type: OC_RP_4G1
+- RF:4
+  - Array, Byte Array, Flat KV object: OC_RP_6GX
+  - no feats type: OC_RP_6G1
+
+In addition, the oid generation API provides an optional mechanism for users to provide hints to the
+DAOS libary to control what redundancy method is chosen and what scale of groups to use for the
```
Comment (shimizukko) [inline src/object/README.md:77]:
```suggestion
DAOS library to control what redundancy method is chosen and what scale of groups to use for the
```

File: src/object/README.md [hunk 3]
```diff
@@ -34,11 +34,66 @@ In addition user can register customized object class by
...
+oclass without needing to specify the oclass itself. Those hints will override the auto class
+selection for that particular setting. For example, one could set a redundancy hint for replication
+on an Array object, and DAOS in thise case will select the proper replicated object class instead of
```
Comment (shimizukko) [inline src/object/README.md:80]:
```suggestion
on an Array object, and DAOS in this case will select the proper replicated object class instead of
```

**Review by shimizukko — APPROVED** [review]

File: src/object/README.md (hunk 1 above)
Comment (mchaarawi) [reply src/object/README.md:51]: curious why you thing the grammar here is wrong? details can be used as a verb no?

Comment (shimizukko) [reply src/object/README.md:51]: ah I see.

File: src/object/README.md [hunk 4]
```diff
@@ -34,11 +34,66 @@ In addition user can register customized object class by
...
+redundancy factor (RF), the number of domain (server engines) of the pool, and on the type of object
+being accessed (determined by the feats flag).
+
+The following details how the object class is chosen when no default or hints are provided:
+- RF:0
+  - Array, Byte Array, Flat KV object: OC_SX
+  - no feats type: OC_S1
+- RF:1
+  - Array, Byte Array:
+    - domain_nr >= 10 : OC_EC_8P1GX
```
Comment (ravalsam) [inline src/object/README.md:57]: Do we need to mention the higher range of EC class ?
- domain_nr >= 18 : OC_EC_16P1GX

**Review by ravalsam — APPROVED** [review]
LGTM

File: src/object/README.md (hunk 4 above)
Comment (mchaarawi) [reply src/object/README.md:57]: this is not implemented currently.. as per wangdi's advice, we check up to 10 and use 8Pn if more.. if we need to add that then we probably need a different ticket to update the selection logic and this doc

**Review by shimizukko — APPROVED** [review]

**Pull request merged**